proto/message_contents/ciphertext.proto

```diff
@@ -20,8 +22,24 @@ message Ciphertext {
     // Encryption: AES256-GCM
     // Key derivation function: HKDF-SHA256
     message Aes256gcmHkdfsha256 {
-        bytes hkdf_salt = 1; // 32 bytes
-        bytes gcm_nonce = 2; // 12 bytes
-        bytes payload = 3; // encrypted payload
+        bytes hkdf_salt = 1;  // 32 bytes
+        bytes gcm_nonce = 2;  // 12 bytes
+        bytes payload = 3;    // encrypted payload
     }
 }
+
+// SignedEciesCiphertext represents an ECIES encrypted payload and a signature
+message SignedEciesCiphertext {
+    // Ecies is ciphertext encrypted using ECIES with a MAC
+    message Ecies {
+        bytes ephemeral_public_key = 1;  // 65 bytes
+        bytes iv = 2;                    // 16 bytes
+        bytes mac = 3;                   // 32 bytes
+        bytes ciphertext = 4;  // encrypted payload with block size of 16
+    }
+
+    // serialized Ecies message
+    bytes ecies_bytes = 1;
+    // signature of the ecies_bytes
+    xmtp.message_contents.Signature signature = 2;
```

Comment:
Looks good, we should document the digest algorithm used for this (sha256, sha3, etc). I've been marching around with my "no signatures without context" banner for a while, but I think those changes can be bundled in later. We're exploring Proofs at the moment with the SendKey work.

Reply:
how exactly will this be signed? Wallet Signature?

Reply:
Good call out. I'll add some clearer docs. 

This is an IdentityKey signature, since it needs to happen in the background (new signature happens every time a conversation is added to the store). 

I thought about using preKeys, but then we would need to include metadata to tell you which key to verify with.

Reply:
`preKey` is an option. We'd just have to reverse out the public key from the signature and find the right one